integration/spark/src/main/java/openlineage/spark/agent/lifecycle/plan/PlanUtils.java

```diff
@@ -92,104 +90,100 @@ public static String namespaceUri(URI outputPath) {
   }
 
   /**
-   * Given a {@link URI}, construct a valid {@link LineageEvent.Dataset} following the expected
+   * Given a {@link URI}, construct a valid {@link OpenLineage.Dataset} following the expected
    * naming conventions.
    *
    * @param outputPath
    * @param schema
    * @return
    */
-  public static LineageEvent.Dataset getDataset(URI outputPath, StructType schema) {
+  public static OpenLineage.Dataset getDataset(URI outputPath, StructType schema) {
     String namespace = namespaceUri(outputPath);
-    LineageEvent.DatasetFacet datasetFacet = datasetFacet(schema, namespace);
-    return getDataset(outputPath, namespace, datasetFacet);
+    OpenLineage.DatasetFacets datasetFacet = datasetFacet(schema, namespace);
+    return getDataset(outputPath.getPath(), namespace, datasetFacet);
   }
 
   /**
    * Construct a dataset given a {@link URI}, namespace, and preconstructed {@link
```

Comment:
Make `dataset` a `{@link Openlineage.Dataset}`. I think it's worth pointing out here that the dataset is neither an `InputDataset` nor an `OutputDataset`, but an anonymous instance since we don't know if we're creating inputs or outputs yet.